packetbeat/protos/mongodb/mongodb.go

```diff
@@ -251,74 +251,65 @@ func (mongodb *Mongodb) onResponse(conn *mongodbConnectionData, msg *mongodbMess
 	mongodb.responses.Put(key, msg)
 }
 
-func (mongodb *Mongodb) onTransComplete(requ, resp *mongodbMessage) {
+func (mongodb *mongodbPlugin) onTransComplete(requ, resp *mongodbMessage) {
 	trans := newTransaction(requ, resp)
-	debugf("Mongodb transaction completed: %s", trans.Mongodb)
+	debugf("Mongodb transaction completed: %s", trans.mongodb)
 
 	mongodb.publishTransaction(trans)
 }
 
 func newTransaction(requ, resp *mongodbMessage) *transaction {
-	trans := &transaction{Type: "mongodb"}
+	trans := &transaction{}
 
 	// fill request
 	if requ != nil {
-		trans.tuple = requ.TCPTuple
-
-		trans.Mongodb = common.MapStr{}
+		trans.mongodb = common.MapStr{}
 		trans.event = requ.event
 		trans.method = requ.method
 
-		trans.cmdline = requ.CmdlineTuple
-		trans.ts = requ.Ts
-		trans.Ts = int64(trans.ts.UnixNano() / 1000) // transactions have microseconds resolution
-		trans.JsTs = requ.Ts
-		trans.Src = common.Endpoint{
-			IP:   requ.TCPTuple.SrcIP.String(),
-			Port: requ.TCPTuple.SrcPort,
-			Proc: string(requ.CmdlineTuple.Src),
+		trans.cmdline = requ.cmdlineTuple
+		trans.ts = requ.ts
+		trans.src = common.Endpoint{
+			IP:   requ.tcpTuple.SrcIP.String(),
+			Port: requ.tcpTuple.SrcPort,
+			Proc: string(requ.cmdlineTuple.Src),
 		}
-		trans.Dst = common.Endpoint{
-			IP:   requ.TCPTuple.DstIP.String(),
-			Port: requ.TCPTuple.DstPort,
-			Proc: string(requ.CmdlineTuple.Dst),
+		trans.dst = common.Endpoint{
+			IP:   requ.tcpTuple.DstIP.String(),
+			Port: requ.tcpTuple.DstPort,
+			Proc: string(requ.cmdlineTuple.Dst),
 		}
-		if requ.Direction == tcp.TCPDirectionReverse {
-			trans.Src, trans.Dst = trans.Dst, trans.Src
+		if requ.direction == tcp.TCPDirectionReverse {
+			trans.src, trans.dst = trans.dst, trans.src
 		}
 		trans.params = requ.params
 		trans.resource = requ.resource
-		trans.BytesIn = requ.messageLength
+		trans.bytesIn = requ.messageLength
 	}
 
 	// fill response
 	if resp != nil {
-		if requ == nil {
```

Comment:
the value being set has been remove from trans, as field trans.tuple was unused.